mlflow/transformers.py

```diff
@@ -990,3 +1110,291 @@ def from_dict(
         )
 
         return _TransformersModel(model, tokenizer, feature_extractor, image_processor, processor)
+
+
+def _get_inference_config(local_path):
+    """
+    Load the inference config if it was provided for use in the `_TransformersWrapper` pyfunc
+    Model wrapper.
+    """
+    config_path = local_path.joinpath("inference_config.txt")
+    if config_path.exists():
+        return json.loads(config_path.read_text())
+
+
+def _load_pyfunc(path):
+    """
+    Loads the model as pyfunc model
+    """
+    local_path = pathlib.Path(path)
+    flavor_configuration = _get_flavor_configuration(local_path, FLAVOR_NAME)
+    inference_config = _get_inference_config(local_path)
+    return _TransformersWrapper(
+        _load_model(str(local_path), flavor_configuration, "pipeline"), inference_config
+    )
+
+
+class _TransformersWrapper:
+    def __init__(self, pipeline, inference_config):
+        self.pipeline = pipeline
+        self.inference_config = inference_config
+        if inference_config:
+            self._validate_inference_config_keys(pipeline, inference_config)
+        self._conversation = None
+
+    @staticmethod
+    def _validate_inference_config_keys(pipeline, inference_config):
+        """
+        Raise an Exception if invalid inference_config keys for the given Pipeline and
+        Model were provided during save.
+        """
+        invalid_keys = []
+        for key in inference_config.keys():
+            if not hasattr(pipeline, key) and not hasattr(pipeline.model, key):
+                invalid_keys.append(key)
+        if invalid_keys:
+            raise MlflowException(
+                "The Inference Configuration overrides that were saved with "
+                "this model are not compatible as overrides to either the "
+                "Pipeline or the Model. Please re-save the model with valid "
+                f"inference configuration keys. Invalid keys: {invalid_keys}"
+            )
+
+    def predict(self, data):
+        import transformers
+
+        if isinstance(data, pd.DataFrame):
+            input_data = data.to_dict(orient="records")
+            if isinstance(self.pipeline, transformers.FillMaskPipeline):
+                # NB: Some pipeline types do not accept a dict input and require either
+                # a str or a List[str]
+                input_data = [list(entry.values())[0] for entry in input_data]
+        elif isinstance(data, dict):
+            input_data = data
+        elif isinstance(data, list):
+            if not all(isinstance(entry, (str, dict)) for entry in data):
+                raise MlflowException(
+                    "Invalid data submission. Ensure all elements in the list are strings."
```

Comment:
I think this error message is missing the dict case:

```suggestion
                    "Invalid data submission. Ensure all elements in the list are strings or dictionaries."
```

Can we also specify what the dictionaries should contain, if we know? Are they string -> string? Can they be something else?

If the dicts can contain keys / values that *aren't* strings, do we need to broaden our signature / schema enforcement support? Looks like the currently proposed changes only handle `List[Dict[str, str]]`.

Reply:
update the comment